### PR TITLE
[14.0][ADD] agreement_legal_exception

### PR DESCRIPTION
**Pull request by kittiu**
- [x] Test script

### CODE REVIEW

**Comment by rousseldenis** [comment]
@kittiu Do you plan to finish this and add tests ?

**Comment by kittiu** [comment]
> @kittiu Do you plan to finish this and add tests ?

Yes

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by newtratip — APPROVED** [review]
LGTM :+1:

File: agreement_legal_exception/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2017 Akretion (http://www.akretion.com)
+# Mourad EL HADJ MIMOUNE <mourad.elhadj.mimoune@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Agreement Exception",
+    "summary": "Custom exceptions on agreement",
+    "version": "14.0.1.0.0",
+    "category": "Contract",
+    "author": "Akretion, Odoo Community Association (OCA)",
```
Comment (Saran440) [inline agreement_legal_exception/__manifest__.py]: Should author and Copyright on header are Ecosoft?

**Review by Saran440 — APPROVED** [review]

File: agreement_legal_exception/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Ecosoft (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Agreement Exception",
+    "summary": "Custom exceptions on agreement",
+    "version": "14.0.1.0.0",
+    "category": "Contract",
+    "author": "Akretion, Odoo Community Association (OCA)",
```
Comment (rousseldenis) [inline agreement_legal_exception/__manifest__.py]: @kittiu Is this real Akretion ?

File: agreement_legal_exception/models/agreement.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+from odoo import api, models
+
+
+class Agreement(models.Model):
+    _inherit = ["agreement", "base.exception"]
+    _name = "agreement"
+    _order = "main_exception_id asc, start_date desc, name desc"
+
+    @api.model
+    def test_all_draft_agreement(self):
```
Comment (rousseldenis) [inline agreement_legal_exception/models/agreement.py]: I would have used same naming as under `detect_draft_exceptions` or something like that

**Review by rousseldenis — APPROVED** [review]
Some minor remarks

**Comment by Saran440** [comment]
@rousseldenis fixed. thank you for help review. i see other module use name `test_all_draft_<model>` So, we will use this name function like that. However, I change it to `detect_draft_exceptions` already.

File: agreement_legal_exception/data/agreement_exception_data.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        <field name="code">model.detect_draft_exceptions()</field>
+        <field name="user_id" ref="base.user_root" />
+        <field name="interval_number">20</field>
+        <field name="interval_type">minutes</field>
+        <field name="numbercall">-1</field>
+        <field name="doall" eval="False" />
+        <field name="active" eval="False" />
+    </record>
+
+    <record id="agreement_excep_no_email" model="exception.rule">
```
Comment (etobella) [inline agreement_legal_exception/data/agreement_exception_data.xml:16]: Shouldn't this go to demo?

**Pull request closed without merging**